include/Gaffer/ValuePlug.inl

```diff
@@ -43,28 +43,18 @@ template<typename T>
 boost::intrusive_ptr<const T> ValuePlug::getObjectValue( const IECore::MurmurHash *precomputedHash ) const
 {
 	IECore::ConstObjectPtr value = getValueInternal( precomputedHash );
-	boost::intrusive_ptr<const T> result = IECore::runTimeCast<const T>( value );
-	if( !result )
+	if( value && value->isInstanceOf( T::staticTypeId() ) )
 	{
-		if( !value )
-		{
-			// This is quite a serious error, and it may occur when a calculation has already been cancelled,
-			// which could result in the exception that first cancelled the calculation being displayed, but
-			// not this error - print the error as a message as well as throwing an exception
-			const std::string error = fmt::format(
-				"{} : getValueInternal() returned NULL. This should be impossible, something has gone badly wrong internally to Gaffer.",
-				fullName()
-			);
-			IECore::msg( IECore::Msg::Error, "ValuePlug::getObjectValue", error );
-			throw IECore::Exception( error );
-		}
```

Comment:
It's been quite a while, and I don't clearly remember all this stuff - but it sounds like this warning was put here quite explicitly to help debug something nasty ... removing it doesn't seem related to this PR, but maybe you're aware of some reason why this logic no longer applies?

Reply:
I traced this back to the original PR, but it still wasn't completely clear to me why this was done in the first place. One thing was clear - there was a segfault in the unexpected `value == nullptr` case that was being fixed in addition to the message being added. I've kept that using the ternary in the format call. The other thing that was clear in the original PR was that the cause of the `value == nullptr` has been fixed elsewhere and has its own test coverage (at the TaskMutex level). So I don't think this message is any more likely to be useful than a belt-and-braces message anywhere else we throw an exception, and since we don't do that as a matter of course, I don't really see a reason to keep it here.

Reply:
OK, I don't feel too strongly about it.